providers/resteasy-validator-provider-11/src/main/java/org/jboss/resteasy/plugins/validation/SimpleViolationsContainer.java

```diff
@@ -25,4 +37,314 @@ public SimpleViolationsContainer(Set<ConstraintViolation<Object>> cvs)
    {
       super(cvs);
    }
+   
+   /**
+    * If some ConstraintViolations are created by Resteasy and some are created by CDI, two
+    * essentially identical ones might appear to be different. For example, those created by Resteasy
+    * might refer to CDI proxy classes while those created by CDI might refer to the backing java classes.
+    */
+   public void addViolations(Set<ConstraintViolation<Object>> cvs)
+   {
+      if (!ejbsPresent(cvs) || getViolations().size() == 0)
+      {
+          getViolations().addAll(cvs); 
+         return;
+      }
+      boolean skip = false;
+      for (ConstraintViolation<Object> cv1 : cvs)
+      {
+         skip = false;
+         for (ConstraintViolation<Object> cv2 : getViolations())
+         {
+            if (compareConstraintViolation(cv1, cv2))
+            {
+               skip = true;
+               break;
+            }
+         }
+         if (!skip)
+         {
+             getViolations().add(cv1);
+         }
+      }
+   }
+
+   public static boolean compareConstraintViolation(ConstraintViolation<?> cv1, ConstraintViolation<?> cv2)
```

Comment:
@ronsigal does this really need to be 'public'?

Reply:
Nope. Thanks.